### PR TITLE
Porting Points docs for extending Mautic.

### PR DESCRIPTION
**Pull request by shinde-rahul**
Adding doc for https://github.com/mautic/developer-documentation-new/issues/66

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/79"></a>

### CODE REVIEW

File: docs/components/points.rst [hunk 1]
```diff
@@ -1,4 +1,234 @@
...
 ######
 
+.. vale off
 
+Point Actions
+*************
+
+.. vale on
+
+A custom Point Action used to give a Contact `x` Points for doing a certain action.
```
Comment (RCheesley) [inline docs/components/points.rst:11]:
```suggestion
In Mautic, custom Point Actions give a Contact `x` Points for doing a certain action.
```

File: docs/components/points.rst [hunk 2]
```diff
@@ -1,4 +1,234 @@
...
+
+Mautic dispatches the Event ``\Mautic\PointBundle\PointEvents::POINT_ON_BUILD`` for Plugins to register their custom point action. Listeners receive a ``Mautic\PointBundle\Event\PointBuilderEvent`` object. Register the Event using the ``addAction`` method as described below.
```
Comment (RCheesley) [inline docs/components/points.rst:13]:
```suggestion
Mautic dispatches the Event ``\Mautic\PointBundle\PointEvents::POINT_ON_BUILD`` for Plugins to register their custom Point Action. Listeners receive a ``Mautic\PointBundle\Event\PointBuilderEvent`` object. Register the Event using the ``addAction`` method as described below.
```

Let's be consistent with the capitalisation :)

File: docs/components/points.rst [hunk 3]
```diff
@@ -1,4 +1,234 @@
...
+    :param array $action: :ref:`Action definition<Custom Point Action definition>`.
+
+.. php:method:: public getActions()
+
+    :return: Array of registered Actions.
+    :returntype: array
+
+.. vale off
+
+Registering a Custom Point Actions
```
Comment (RCheesley) [inline docs/components/points.rst:29]:
```suggestion
Registering a Custom Point Action
```

File: docs/components/points.rst [hunk 4]
```diff
@@ -1,4 +1,234 @@
...
+      - Static callback function used to validate the action. Return true to add the Points to the Contact.
+
+.. vale off
+
+Point Triggers
+**************
+
+.. vale on
+
+A custom Point Trigger used to execute a specific action once a Contact hits X number of Points.
```
Comment (RCheesley) [inline docs/components/points.rst:129]:
```suggestion
A custom Point Trigger used to execute a specific action once a Contact reaches X number of Points.
```

File: docs/components/points.rst [hunk 5]
```diff
@@ -1,4 +1,234 @@
...
+    :param array $action: :ref:`Action definition<Custom Point Trigger definition>`.
+
+.. php:method:: public getEvents()
+
+    :return: Array of registered Events.
+    :returntype: array
+
+.. vale off
+
+Registering a Custom Point Triggers
```
Comment (RCheesley) [inline docs/components/points.rst:147]:
```suggestion
Registering a Custom Point Trigger
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just a few small suggestions - great job and good use of the PHP domain!

**Review by dennisameling — APPROVED** [review]

**Comment by dennisameling** [comment]
Good to merge from my side. @RCheesley mind doing a final check?

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just one duplicated colon to fix

**Review by dennisameling — APPROVED** [review]

**Pull request merged**